# Data-Loading-Ideating-Reiterating

### Topic Exploraion

I have decided to look into measurements of aquatic organism tissue sample results in the state of California. High amounts of heavy metals in aquatic organism often time serves as an indicator of water quality. Originally, I wanted to predict the level of mercury in organism tissues, but I struggled to find a data set that contains both tissue analysis results and water sample collected at the same location on the same date. The first section of this notebook is where you see my work into trying to work out jointing the two datasets. Eventually, I realized that I have to focus on only one of the datasets and see what kind of insights I can gain from there. 

Later, I shifted my focus back on just the tissue data set, and decided investigate further to see if there is anything that could correlate to the expected levels of mercury & other harmful chemicals such as PCBs & PBDEs. Things that has been shown to have affect on these harmful chemical levels within an organism are temperature and location. Some species also tend to accumulate more harmful chemicals than others. Another interesting factor is that selenium level in aquatic life has an affect on Mercury levels. With this dataset, I can make a prediction model on which aquatic species will have the highest levels of mercury & PCBs based on time of the year and their location. It would also be interesting to test for affinity between selenium & mercury levels. 

In [78]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None) #just to makes sure all columns are visible
pd.set_option('display.max_info_rows', 100)
pd.set_option('display.max_rows', None)

### Source Data Gathering

Here is the data source https://data.ca.gov/dataset/surface-water-aquatic-organism-tissue-sample-results.
The files are all trancated in years. I want to take 5 years of data to get some kind of trend so I downloaded the data from 2019-2023. I have to first load each individual files into the notebook, examine them individually, and then concatonate them into a single master file

In [79]:
df23 = pd.read_csv('tissuedata_year-2023_2024-10-01.csv') #2023 data

In [80]:
df23.head()

ProgramName  \
0  Surface Water Ambient Monitoring Program   
1  Surface Water Ambient Monitoring Program   
2  Surface Water Ambient Monitoring Program   
3  Surface Water Ambient Monitoring Program   
4  Surface Water Ambient Monitoring Program   

                          ParentProjectName            ProjectCode  \
0  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
1  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
3  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
4  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   

                                 ProjectName           CompositeID  \
0  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
1  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
2  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
3  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
4  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   

        StationName StationCode   Latitude   Longitude GeometryShape  \
0  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
1  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
2  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
3  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
4  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   

  EarliestDateSampled CompositeType                CommonName  \
0          04/19/2023        normal  California Spiny Lobster   
1          04/19/2023        normal  California Spiny Lobster   
2          04/19/2023        normal  California Spiny Lobster   
3          04/19/2023        normal  California Spiny Lobster   
4          04/19/2023        normal  California Spiny Lobster   

                 FinalID TissuePrep               TissueName  \
0  Panulirus interruptus        NaN  whole organism no shell   
1  Panulirus interruptus        NaN  whole organism no shell   
2  Panulirus interruptus        NaN  whole organism no shell   
3  Panulirus interruptus        NaN  whole organism no shell   
4  Panulirus interruptus        NaN  whole organism no shell   

   NumberFishperComp SampleTypeCode  CompositeReplicate  ResultReplicate  \
0                4.0      Composite                 1.0              1.0   
1                4.0      Composite                 1.0              1.0   
2                4.0      Composite                 1.0              1.0   
3                4.0      Composite                 1.0              1.0   
4                4.0      Composite                 1.0              1.0   

   Matrix      Method   Analyte     Unit  Result ResQualCode   MDL    RL  \
0  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
1  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
2  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
3  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
4  tissue  EPA 200.8m  Moisture        %   75.50           =   NaN   NaN   

  QACode BatchVerification ComplianceCode  DilutionFactor LabSampleID  \
0    NaN               VAC            Com             1.0  C122OCHLOB   
1    NaN               VAC            Com             1.0  C122OCHLOB   
2    NaN               VAC            Com             1.0  C122OCHLOB   
3    NaN               VAC            Com             1.0  C122OCHLOB   
4    NaN               VAC            Com             1.0  C122OCHLOB   

  ResultComments PrepPreservationName PrepPreservationDate  \
0            NaN          FieldFrozen           04/19/2023   
1            NaN          FieldFrozen           04/19/2023   
2            NaN          FieldFrozen           04/19/2023   
3            NaN          FieldFrozen           04/19/2023   
4            NaN          FieldFroz

In [81]:
df22 = pd.read_csv('tissuedata_year-2022_2024-10-01.csv')
df22.head()

ProgramName ParentProjectName  \
0  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
1  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
2  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
3  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
4  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   

       ProjectCode      ProjectName                CompositeID  \
0  CCLEAN_2022_Wet  CCLEAN_2022_Wet  C_CarmRiv1-Feb2022_CCLEAN   
1  CCLEAN_2022_Wet  CCLEAN_2022_Wet  C_CarmRiv1-Feb2022_CCLEAN   
2  CCLEAN_2022_Wet  CCLEAN_2022_Wet  C_CarmRiv1-Feb2022_CCLEAN   
3  CCLEAN_2022_Wet  CCLEAN_2022_Wet  C_CarmRiv1-Feb2022_CCLEAN   
4  CCLEAN_2022_Wet  CCLEAN_2022_Wet  C_CarmRiv1-Feb2022_CCLEAN   

                  StationName StationCode  Latitude  Longitude GeometryShape  \
0  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
1  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
2  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
3  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
4  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   

  EarliestDateSampled CompositeType         CommonName                FinalID  \
0          02/28/2022        Normal  California Mussel  Mytilus californianus   
1          02/28/2022        Normal  California Mussel  Mytilus californianus   
2          02/28/2022        Normal  California Mussel  Mytilus californianus   
3          02/28/2022        Normal  California Mussel  Mytilus californianus   
4          02/28/2022        Normal  California Mussel  Mytilus californianus   

  TissuePrep                                   TissueName  NumberFishperComp  \
0        NaN  soft tissue (e.g. clams) with gonads intact               45.0   
1        NaN  soft tissue (e.g. clams) with gonads intact               45.0   
2        NaN  soft tissue (e.g. clams) with gonads intact               45.0   
3        NaN  soft tissue (e.g. clams) with gonads intact               45.0   
4        NaN  soft tissue (e.g. clams) with gonads intact               45.0   

  SampleTypeCode  CompositeReplicate  ResultReplicate  Matrix  \
0      Composite                 1.0              1.0  tissue   
1      Composite                 1.0              1.0  tissue   
2      Composite                 1.0              1.0  tissue   
3      Composite                 1.0              1.0  tissue   
4      Composite                 1.0              1.0  tissue   

             Method             Analyte        Unit    Result ResQualCode  \
0  SGS AXYS MLA-210               Lipid        % ww   0.89000           =   
1  SGS AXYS MLA-210            Moisture        % ww  87.10000           =   
2  SGS AXYS MLA-210             PCB 001    ug/Kg dw   0.00253           =   
3  SGS AXYS MLA-210  PCB 001(Surrogate)  % recovery  20.10000           =   
4  SGS AXYS MLA-210             PCB 002    ug/Kg dw   0.00296           =   

        MDL       RL QACode BatchVerification ComplianceCode  DilutionFactor  \
0       NaN      NaN    NaN               VLF            Com             1.0   
1       NaN      NaN    NaN               VLF            Com             1.0   
2  0.000181  0.00362     SC               VLF            Com             1.0   
3       NaN      NaN    VGN               VLF            Com             1.0   
4  0.000181  0.00358     SC               VLF            Com             1.0   

  LabSampleID       ResultComments PrepPreservationName PrepPreservationDate  \
0  L36676-6 i                  NaN         Not Recorded           01/01/1950   
1  L36676-6 i                  NaN         Not Recorded           01/01/1950   
2  L36676-6 i  surrogate corrected         Not Recorded           01/01/1950   
3  L36676-6 i                  NaN         Not Recorded           01/01/1950   
4  L36676-6 i  

In [82]:
df21 = pd.read_csv('tissuedata_year-2021_2024-10-01.csv')
df21.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/3262558734.py:1: DtypeWarning: Columns (33,95,96,100,108,113) have mixed types. Specify dtype option on import or set low_memory=False.
  df21 = pd.read_csv('tissuedata_year-2021_2024-10-01.csv')


ProgramName  \
0  Delta Regional Monitoring Program   
1  Delta Regional Monitoring Program   
2  Delta Regional Monitoring Program   
3  Delta Regional Monitoring Program   
4  Delta Regional Monitoring Program   

                              ParentProjectName ProjectCode  \
0  Delta RMP - Constituents of Emerging Concern  21DRMP5CEC   
1  Delta RMP - Constituents of Emerging Concern  21DRMP5CEC   
2  Delta RMP - Constituents of Emerging Concern  21DRMP5CEC   
3  Delta RMP - Constituents of Emerging Concern  21DRMP5CEC   
4  Delta RMP - Constituents of Emerging Concern  21DRMP5CEC   

                                       ProjectName  \
0  2021 Delta RMP Constituents of Emerging Concern   
1  2021 Delta RMP Constituents of Emerging Concern   
2  2021 Delta RMP Constituents of Emerging Concern   
3  2021 Delta RMP Constituents of Emerging Concern   
4  2021 Delta RMP Constituents of Emerging Concern   

                     CompositeID  \
0  DRMP-CEC-510SACC3A-2021-10-21   
1  DRMP-CEC-510SACC3A-2021-10-21   
2  DRMP-CEC-510SACC3A-2021-10-21   
3  DRMP-CEC-510SACC3A-2021-10-21   
4  DRMP-CEC-510SACC3A-2021-10-21   

                                         StationName StationCode  Latitude  \
0  Sacramento River at Hood Monitoring Station Pl...   510SACC3A  38.36771   
1  Sacramento River at Hood Monitoring Station Pl...   510SACC3A  38.36771   
2  Sacramento River at Hood Monitoring Station Pl...   510SACC3A  38.36771   
3  Sacramento River at Hood Monitoring Station Pl...   510SACC3A  38.36771   
4  Sacramento River at Hood Monitoring Station Pl...   510SACC3A  38.36771   

   Longitude GeometryShape EarliestDateSampled CompositeType       CommonName  \
0  -121.5205         Point          10/21/2021        Normal  Freshwater Clam   
1  -121.5205         Point          10/21/2021        Normal  Freshwater Clam   
2  -121.5205         Point          10/21/2021        Normal  Freshwater Clam   
3  -121.5205         Point          10/21/2021        Normal  Freshwater Clam   
4  -121.5205         Point          10/21/2021        Normal  Freshwater Clam   

              FinalID   TissuePrep  \
0  Corbicula fluminea  FieldFrozen   
1  Corbicula fluminea  FieldFrozen   
2  Corbicula fluminea  FieldFrozen   
3  Corbicula fluminea  FieldFrozen   
4  Corbicula fluminea  FieldFrozen   

                                    TissueName  NumberFishperComp  \
0  soft tissue (e.g. clams) with gonads intact              400.0   
1  soft tissue (e.g. clams) with gonads intact              400.0   
2  soft tissue (e.g. clams) with gonads intact              400.0   
3  soft tissue (e.g. clams) with gonads intact              400.0   
4  soft tissue (e.g. clams) with gonads intact              400.0   

  SampleTypeCode  CompositeReplicate  ResultReplicate  Matrix  \
0      Composite                 1.0              1.0  tissue   
1      Composite                 1.0              1.0  tissue   
2      Composite                 1.0              1.0  tissue   
3      Composite                 1.0              1.0  tissue   
4      Composite                 1.0              1.0  tissue   

                Method   Analyte  Unit  Result ResQualCode  MDL  RL QACode  \
0  AXYS MLA-033 Rev 06  Moisture  % ww    88.5           =  NaN NaN    NaN   
1  AXYS MLA-033 Rev 06  Moisture  % ww    88.5           =  NaN NaN    NaN   
2  AXYS MLA-033 Rev 06  Moisture  % ww    88.5           =  NaN NaN    NaN   
3  AXYS MLA-033 Rev 06  Moisture  % ww    88.5           =  NaN NaN    NaN   
4  AXYS MLA-033 Rev 06  Moisture  % ww    88.5           =  NaN NaN    NaN   

  BatchVerification ComplianceCode  DilutionFactor LabSampleID ResultComments  \
0           VAC,VQI            Com             1.0    L36293-5            NaN   
1           VAC,VQI            Com             1.0    L36293-5            NaN   
2           VAC,VQI            Com             1.0    L36293-5            NaN   
3           VAC,VQI            Com             1.0    L36293-5            NaN  

In [83]:
list =[33,95,96,100,108,113]
column = df21.iloc[0:,list]# To address the Dtype address, I want to look at the column headers to see if it's info that I need
column.head()

ResultComments Anomaly ProcessedOrganismsExpandedComments PartsComments  \
0            NaN     NaN                                NaN           NaN   
1            NaN     NaN                                NaN           NaN   
2            NaN     NaN                                NaN           NaN   
3            NaN     NaN                                NaN           NaN   
4            NaN     NaN                                NaN           NaN   

        TissueCollectionComments SampleID  
0  Collection end time at 15:35.      NaN  
1  Collection end time at 15:35.      NaN  
2  Collection end time at 15:35.      NaN  
3  Collection end time at 15:35.      NaN  
4  Collection end time at 15:35.      NaN

These are columns that I will drop later for analysis, so I will just leave them for now. 

In [84]:
df20 = pd.read_csv('tissuedata_year-2020_2024-10-01.csv')
df20.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/2073405848.py:1: DtypeWarning: Columns (33,77,78,95,96,100,107,108,109,113) have mixed types. Specify dtype option on import or set low_memory=False.
  df20 = pd.read_csv('tissuedata_year-2020_2024-10-01.csv')


ProgramName ParentProjectName  \
0  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
1  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
2  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
3  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   
4  Central Coast Long term Environmental Assessme...       RWB3 CCLEAN   

       ProjectCode      ProjectName                CompositeID  \
0  CCLEAN_2019_Wet  CCLEAN_2019_Wet  C_CarmRiv1-Apr2020_CCLEAN   
1  CCLEAN_2019_Wet  CCLEAN_2019_Wet  C_CarmRiv1-Apr2020_CCLEAN   
2  CCLEAN_2019_Wet  CCLEAN_2019_Wet  C_CarmRiv1-Apr2020_CCLEAN   
3  CCLEAN_2019_Wet  CCLEAN_2019_Wet  C_CarmRiv1-Apr2020_CCLEAN   
4  CCLEAN_2019_Wet  CCLEAN_2019_Wet  C_CarmRiv1-Apr2020_CCLEAN   

                  StationName StationCode  Latitude  Longitude GeometryShape  \
0  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
1  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
2  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
3  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   
4  Carmel River Beach mussels    CarmRiv1    36.539   -121.932         Point   

  EarliestDateSampled CompositeType         CommonName                FinalID  \
0          04/06/2020        Normal  California Mussel  Mytilus californianus   
1          04/06/2020        Normal  California Mussel  Mytilus californianus   
2          04/06/2020        Normal  California Mussel  Mytilus californianus   
3          04/06/2020        Normal  California Mussel  Mytilus californianus   
4          04/06/2020        Normal  California Mussel  Mytilus californianus   

  TissuePrep                                   TissueName  NumberFishperComp  \
0        NaN  soft tissue (e.g. clams) with gonads intact               49.0   
1        NaN  soft tissue (e.g. clams) with gonads intact               49.0   
2        NaN  soft tissue (e.g. clams) with gonads intact               49.0   
3        NaN  soft tissue (e.g. clams) with gonads intact               49.0   
4        NaN  soft tissue (e.g. clams) with gonads intact               49.0   

  SampleTypeCode  CompositeReplicate  ResultReplicate  Matrix  \
0      Composite                 1.0              1.0  tissue   
1      Composite                 1.0              1.0  tissue   
2      Composite                 1.0              1.0  tissue   
3      Composite                 1.0              1.0  tissue   
4      Composite                 1.0              1.0  tissue   

                Method             Analyte        Unit    Result ResQualCode  \
0  AXYS MLA-010 Rev 12               Lipid        % ww   1.53000           =   
1  AXYS MLA-010 Rev 12            Moisture        % ww  86.00000           =   
2  AXYS MLA-010 Rev 12             PCB 001    ug/Kg dw   0.00186           =   
3  AXYS MLA-010 Rev 12  PCB 001(Surrogate)  % recovery  49.80000           =   
4  AXYS MLA-010 Rev 12             PCB 002    ug/Kg dw   0.00220           =   

        MDL      RL QACode BatchVerification ComplianceCode  DilutionFactor  \
0       NaN     NaN    NaN               VLF             NR             1.0   
1       NaN     NaN    NaN               VLF             NR             1.0   
2  0.000513  0.0308     SC               VLF             NR             1.0   
3       NaN     NaN    VGN               VLF             NR             1.0   
4  0.000513  0.0308     SC               VLF             NR             1.0   

  LabSampleID       ResultComments PrepPreservationName PrepPreservationDate  \
0    L33059-6                  NaN         Not Recorded           01/01/1950   
1    L33059-6                  NaN         Not Recorded           01/01/1950   
2    L33059-6  surrogate corrected         Not Recorded           01/01/1950   
3    L33059-6                  NaN         Not Recorded           01/01/1950   
4  

In [85]:
list =[33,77,78,95,96,100,107,108,109,113]
column = df20.iloc[0:,list]# To address the Dtype address, I want to look at the column headers to see if it's info that I need
column.head()

ResultComments HydroModLoc Hydromod Anomaly  \
0                  NaN         NaN      NaN     NaN   
1                  NaN         NaN      NaN     NaN   
2  surrogate corrected         NaN      NaN     NaN   
3                  NaN         NaN      NaN     NaN   
4  surrogate corrected         NaN      NaN     NaN   

  ProcessedOrganismsExpandedComments PartsComments SampleComments  \
0                                NaN           NaN            NaN   
1                                NaN           NaN            NaN   
2                                NaN           NaN            NaN   
3                                NaN           NaN            NaN   
4                                NaN           NaN            NaN   

  TissueCollectionComments LocationDetailTIComments SampleID  
0                      NaN                      NaN      NaN  
1                      NaN                      NaN      NaN  
2                      NaN                      NaN      NaN  
3                      NaN                      NaN      NaN  
4                      NaN                      NaN      NaN

Also columns that I will drop later for analysis, so I will just leave them for now. 

In [86]:
df19 = pd.read_csv('tissuedata_year-2019_2024-10-01.csv')
df19.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/2588635892.py:1: DtypeWarning: Columns (14,33,39,70,77,78,84,87,89,90,91,93,94,95,96,100,107,109,113) have mixed types. Specify dtype option on import or set low_memory=False.
  df19 = pd.read_csv('tissuedata_year-2019_2024-10-01.csv')


ProgramName  \
0  Colorado Lagoon TMDL Compliance Monitoring   
1  Colorado Lagoon TMDL Compliance Monitoring   
2  Colorado Lagoon TMDL Compliance Monitoring   
3  Colorado Lagoon TMDL Compliance Monitoring   
4  Colorado Lagoon TMDL Compliance Monitoring   

                            ParentProjectName ProjectCode  \
0  Colorado Lagoon TMDL Compliance Monitoring     19CLSmr   
1  Colorado Lagoon TMDL Compliance Monitoring     19CLSmr   
2  Colorado Lagoon TMDL Compliance Monitoring     19CLSmr   
3  Colorado Lagoon TMDL Compliance Monitoring     19CLSmr   
4  Colorado Lagoon TMDL Compliance Monitoring     19CLSmr   

                                         ProjectName CompositeID  \
0  City of Long Beach Colorado Lagoon TMDL Monito...   F1-190709   
1  City of Long Beach Colorado Lagoon TMDL Monito...   F1-190709   
2  City of Long Beach Colorado Lagoon TMDL Monito...   F1-190709   
3  City of Long Beach Colorado Lagoon TMDL Monito...   F1-190709   
4  City of Long Beach Colorado Lagoon TMDL Monito...   F1-190709   

                                         StationName StationCode   Latitude  \
0  Composite of 3 sub areas within the Western Ar...          F1  33.771563   
1  Composite of 3 sub areas within the Western Ar...          F1  33.771563   
2  Composite of 3 sub areas within the Western Ar...          F1  33.771563   
3  Composite of 3 sub areas within the Western Ar...          F1  33.771563   
4  Composite of 3 sub areas within the Western Ar...          F1  33.771563   

    Longitude GeometryShape EarliestDateSampled CompositeType CommonName  \
0 -118.133022     PointCent          07/09/2019        Normal  Top Smelt   
1 -118.133022     PointCent          07/09/2019        Normal  Top Smelt   
2 -118.133022     PointCent          07/09/2019        Normal  Top Smelt   
3 -118.133022     PointCent          07/09/2019        Normal  Top Smelt   
4 -118.133022     PointCent          07/09/2019        Normal  Top Smelt   

              FinalID    TissuePrep      TissueName  NumberFishperComp  \
0  Atherinops affinis  Not Recorded  whole organism               26.0   
1  Atherinops affinis  Not Recorded  whole organism               26.0   
2  Atherinops affinis  Not Recorded  whole organism               26.0   
3  Atherinops affinis  Not Recorded  whole organism               26.0   
4  Atherinops affinis  Not Recorded  whole organism               26.0   

  SampleTypeCode  CompositeReplicate  ResultReplicate  Matrix         Method  \
0      Composite                 1.0              1.0  tissue  EPA 8270C SIM   
1      Composite                 1.0              1.0  tissue  EPA 8270C SIM   
2      Composite                 1.0              1.0  tissue  EPA 8270C SIM   
3      Composite                 1.0              1.0  tissue  EPA 8270C SIM   
4      Composite                 1.0              1.0  tissue  EPA 8270C SIM   

             Analyte      Unit  Result ResQualCode  MDL    RL QACode  \
0     Acenaphthylene  ug/Kg ww     NaN          ND  1.8  10.0    NaN   
1     Acenaphthylene  ug/Kg ww     NaN          ND  1.8  10.0    NaN   
2         Anthracene  ug/Kg ww     NaN          ND  3.5  10.0    NaN   
3         Anthracene  ug/Kg ww     NaN          ND  3.5  10.0    NaN   
4  Benz(a)anthracene  ug/Kg ww     NaN          ND  2.1  10.0    NaN   

  BatchVerification ComplianceCode  DilutionFactor LabSampleID ResultComments  \
0                NR            Com             1.0  570-1214-1            NaN   
1                NR            Com             1.0  570-1214-1            NaN   
2                NR            Com             1.0  570-1214-1            NaN   
3                NR            Com             1.0  570-1214-1            NaN   
4                NR            Com             1.0  570-1214-1            NaN   

  PrepPreservationName PrepPreservationDate DigestExtractMethod  \
0         Not Recorded           01/01/1950            EPA 3541   
1         Not Recorded           01/01/1950        

In [87]:
list =[14,33,39,70,77,78,84,87,89,90,91,93,94,95,96,100,107,109,113]
column = df19.iloc[0:,list]# To address the Dtype address, I want to look at the column headers to see if it's info that I need
column.head()

TissuePrep ResultComments  \
0  Not Recorded            NaN   
1  Not Recorded            NaN   
2  Not Recorded            NaN   
3  Not Recorded            NaN   
4  Not Recorded            NaN   

                                   CompositeComments CompositeTissuePrep  \
0  Composite of topsmelt collected at F1a, F1b an...        Not Recorded   
1  Composite of topsmelt collected at F1a, F1b an...        Not Recorded   
2  Composite of topsmelt collected at F1a, F1b an...        Not Recorded   
3  Composite of topsmelt collected at F1a, F1b an...        Not Recorded   
4  Composite of topsmelt collected at F1a, F1b an...        Not Recorded   

  HydroModLoc Hydromod TagNumber LengthSource UnitLengthFish UnitWeightFish  \
0         NaN      NaN       NaN        Field             cm            NaN   
1         NaN      NaN       NaN        Field             cm            NaN   
2         NaN      NaN       NaN        Field             cm            NaN   
3         NaN      NaN       NaN        Field             cm            NaN   
4         NaN      NaN       NaN        Field             cm            NaN   

  WeightSource  Age  Sex Anomaly ProcessedOrganismsExpandedComments  \
0        Field  NaN  Unk     NaN                                NaN   
1        Field  NaN  Unk     NaN                                NaN   
2        Field  NaN  Unk     NaN                                NaN   
3        Field  NaN  Unk     NaN                                NaN   
4        Field  NaN  Unk     NaN                                NaN   

  PartsComments SampleComments LocationDetailTIComments   SampleID  
0           NaN            NaN                      NaN  F1-190709  
1           NaN            NaN                      NaN  F1-190709  
2           NaN            NaN                      NaN  F1-190709  
3           NaN            NaN                      NaN  F1-190709  
4           NaN            NaN                      NaN  F1-190709

The columns 'UnitLengthFish, UnitWeightFish, Age, Sex' may be useful, I need to take a deeper look at them and see what's there

In [88]:
#Look at UnitLenghtFish values first
df19['UnitLengthFish'].value_counts()

UnitLengthFish
mm    30912
cm      366
Name: count, dtype: int64

In [89]:
#Look at UnitLenghtFish nulls
df19['UnitLengthFish'].isnull().sum()

45368

In [90]:
df19['UnitLengthFish'].dtype

dtype('O')

The mixed dtype is probably due to the null values saved as a different dtype than the values. Nulls are usually floats in Pandas. Leaving it now since I will be processing null values later for the master df. 

In [91]:
#Look at UnitWeightFish values first
df19['UnitWeightFish'].value_counts()

UnitWeightFish
g    30767
Name: count, dtype: int64

In [92]:
#Look at UnitWeightFish nulls
df19['UnitWeightFish'].isnull().sum()

45879

In [93]:
df19['UnitWeightFish'].dtype

dtype('O')

Same as above, mixed dtype due to the nulls.

In [94]:
#Look at Age values first
df19['Age'].value_counts()

Age
NR    28635
Name: count, dtype: int64

In [95]:
#Look at Age nulls
df19['Age'].isnull().sum()

48011

In [96]:
df19['Age'].dtype

dtype('O')

Age actually doesn't look like it's useful here. Will drop this column later

In [97]:
#Look at Sex values first
df19['Sex'].value_counts()

Sex
NR     11345
M       8482
F       6957
LAB     2835
Unk      872
Name: count, dtype: int64

In [98]:
#Look at Sex nulls
df19['Sex'].isnull().sum()

46155

In [99]:
df19['Sex'].dtype

dtype('O')

Same as above, mixed dtype due to the nulls.

Now that I have finished taking a quick look into the columns with mixed dtypes, I want to look at all the dataframes to make sure that they all have the same amount of columns

In [100]:
def multidf_shape(df):
    """prints out the shape of multiple dataframes at once"""

    for i, df in enumerate(df[0:],0): #i= whichever value the function is indexing. df[0:] slices the df at row 1, going from list index 0 
        print(df.shape) #print out the shape of each dataframes

multidf_shape([df19,df20,df21,df22,df23])

(76646, 117)
(67762, 117)
(60158, 117)
(4743, 117)
(7488, 117)


Next, I want to look at the headers to make sure that all 5 dataframes have the same headers

In [101]:
def compare_headers(df): # a function that took me forever to write, but worth it
    """Compares headers across multiple dataframes against base header and print out 'Columns Unmatched' if there is difference
    and 'Columns Match' if there are the same."""

    header_0 = df[0].columns #this takes the header of the first dataframe in the list, which is index 0, and compares the headers of the subsequent dataframes in the list against it
    for i, df in enumerate(df[0:], 1): # the enumerate function reiterates down the list in row 1, starting at index 1
        diff = set(header_0) != set(df.columns) # making headers into a set to compare. Defining the variable "diff". If the sets don't equal, then they are different and gets stored as variable "diff". 
        if diff: # if the comparison between current index and the base index (index 0) is different
            print(f"Headers Unmatched") # if the above condition is met, print "Headers Unmatched"
        else:
            print(f"Headers Match")  # if the above condition is not met, print "Headers Matched"

compare_headers([df19,df20,df21,df22,df23])

Headers Match
Headers Match
Headers Match
Headers Match
Headers Match


By logic, if header 1 = header 0, and all subsequent header all = header 0, then all headers in all dfs match each other. Since they are all the same. We can concatenant into a single master file/dataframe to work with

### Making Master Dataset

In [102]:
df_tissue = pd.concat([df23,df22,df21,df20,df19]) #to combine all 5 datasets into one continuous file
df_tissue.head() # df_tissue is now my temp master dataframe

ProgramName  \
0  Surface Water Ambient Monitoring Program   
1  Surface Water Ambient Monitoring Program   
2  Surface Water Ambient Monitoring Program   
3  Surface Water Ambient Monitoring Program   
4  Surface Water Ambient Monitoring Program   

                          ParentProjectName            ProjectCode  \
0  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
1  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
3  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
4  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   

                                 ProjectName           CompositeID  \
0  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
1  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
2  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
3  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
4  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   

        StationName StationCode   Latitude   Longitude GeometryShape  \
0  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
1  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
2  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
3  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
4  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   

  EarliestDateSampled CompositeType                CommonName  \
0          04/19/2023        normal  California Spiny Lobster   
1          04/19/2023        normal  California Spiny Lobster   
2          04/19/2023        normal  California Spiny Lobster   
3          04/19/2023        normal  California Spiny Lobster   
4          04/19/2023        normal  California Spiny Lobster   

                 FinalID TissuePrep               TissueName  \
0  Panulirus interruptus        NaN  whole organism no shell   
1  Panulirus interruptus        NaN  whole organism no shell   
2  Panulirus interruptus        NaN  whole organism no shell   
3  Panulirus interruptus        NaN  whole organism no shell   
4  Panulirus interruptus        NaN  whole organism no shell   

   NumberFishperComp SampleTypeCode  CompositeReplicate  ResultReplicate  \
0                4.0      Composite                 1.0              1.0   
1                4.0      Composite                 1.0              1.0   
2                4.0      Composite                 1.0              1.0   
3                4.0      Composite                 1.0              1.0   
4                4.0      Composite                 1.0              1.0   

   Matrix      Method   Analyte     Unit  Result ResQualCode   MDL    RL  \
0  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
1  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
2  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
3  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
4  tissue  EPA 200.8m  Moisture        %   75.50           =   NaN   NaN   

  QACode BatchVerification ComplianceCode  DilutionFactor LabSampleID  \
0    NaN               VAC            Com             1.0  C122OCHLOB   
1    NaN               VAC            Com             1.0  C122OCHLOB   
2    NaN               VAC            Com             1.0  C122OCHLOB   
3    NaN               VAC            Com             1.0  C122OCHLOB   
4    NaN               VAC            Com             1.0  C122OCHLOB   

  ResultComments PrepPreservationName PrepPreservationDate  \
0            NaN          FieldFrozen           04/19/2023   
1            NaN          FieldFrozen           04/19/2023   
2            NaN          FieldFrozen           04/19/2023   
3            NaN          FieldFrozen           04/19/2023   
4            NaN          FieldFroz

In [103]:
df_tissue.shape #checking to see if I still all the rows, and if I got all the rows.

(216797, 117)

In [104]:
print(sum([76646, 67762, 60158, 4743, 7488])) # It matches.

216797


In [105]:
df_tissue.info(117) # Just a glance into dtypes

<class 'pandas.core.frame.DataFrame'>
Index: 216797 entries, 0 to 76645
Data columns (total 117 columns):
 #    Column                              Dtype  
---   ------                              -----  
 0    ProgramName                         object 
 1    ParentProjectName                   object 
 2    ProjectCode                         object 
 3    ProjectName                         object 
 4    CompositeID                         object 
 5    StationName                         object 
 6    StationCode                         object 
 7    Latitude                            float64
 8    Longitude                           float64
 9    GeometryShape                       object 
 10   EarliestDateSampled                 object 
 11   CompositeType                       object 
 12   CommonName                          object 
 13   FinalID                             object 
 14   TissuePrep                          object 
 15   TissueName                          ob

In [106]:
df_tissue.isna().sum() # see how many nulls I've got

ProgramName                                0
ParentProjectName                          0
ProjectCode                                0
ProjectName                                0
CompositeID                                0
StationName                                0
StationCode                                0
Latitude                                   0
Longitude                                  0
GeometryShape                          14983
EarliestDateSampled                        0
CompositeType                              0
CommonName                                 0
FinalID                                    0
TissuePrep                             49754
TissueName                                 0
NumberFishperComp                         77
SampleTypeCode                             0
CompositeReplicate                         0
ResultReplicate                            0
Matrix                                     0
Method                                     0
Analyte   

The columns with nulls that I care about the most is 'Result', there is 91569 out of 216797 samples. That is quite significant. Let's look at which analyte has a null result to determine if this dataset is usable

In [107]:
results_null = df_tissue.loc[df_tissue['Result'].isnull()] 
results_null.head() #created another temp dataframe to look at all rows that has Nan in the 'Result' column

ProgramName  \
28   Surface Water Ambient Monitoring Program   
29   Surface Water Ambient Monitoring Program   
30   Surface Water Ambient Monitoring Program   
31   Surface Water Ambient Monitoring Program   
169  Surface Water Ambient Monitoring Program   

                            ParentProjectName            ProjectCode  \
28   SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
29   SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
30   SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
31   SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
169                     SWAMP RWB7 Monitoring   RWB7_Trend_Fall_2023   

                                   ProjectName           CompositeID  \
28   Statewide BMP Realignment RWB9 Study 2022  C1_90606MISSRLN22LOB   
29   Statewide BMP Realignment RWB9 Study 2022  C1_90606MISSRLN22LOB   
30   Statewide BMP Realignment RWB9 Study 2022  C1_90606MISSRLN22LOB   
31   Statewide BMP Realignment RWB9 Study 2022  C1_90606MISSRLN22LOB   
169          RWB7 Trend Monitoring Fall CY2023   C1_723NRBDRYR723TIL   

               StationName StationCode   Latitude   Longitude GeometryShape  \
28             Mission Bay   90606MISS  32.770215 -117.238821   PolygonCent   
29             Mission Bay   90606MISS  32.770215 -117.238821   PolygonCent   
30             Mission Bay   90606MISS  32.770215 -117.238821   PolygonCent   
31             Mission Bay   90606MISS  32.770215 -117.238821   PolygonCent   
169  New River at Boundary   723NRBDRY  32.665830 -115.502220         Point   

    EarliestDateSampled CompositeType                CommonName  \
28           02/15/2023        normal  California Spiny Lobster   
29           02/15/2023        normal  California Spiny Lobster   
30           02/15/2023        normal  California Spiny Lobster   
31           02/15/2023        normal  California Spiny Lobster   
169          10/25/2023        normal              Tilapia spp.   

                   FinalID TissuePrep               TissueName  \
28   Panulirus interruptus        NaN  whole organism no shell   
29   Panulirus interruptus        NaN  whole organism no shell   
30   Panulirus interruptus        NaN  whole organism no shell   
31   Panulirus interruptus        NaN  whole organism no shell   
169                Tilapia   Skin off                   fillet   

     NumberFishperComp SampleTypeCode  CompositeReplicate  ResultReplicate  \
28                 4.0      Composite                 1.0              1.0   
29                 4.0      Composite                 1.0              1.0   
30                 4.0      Composite                 1.0              1.0   
31                 4.0      Composite                 1.0              1.0   
169               11.0      Composite                 1.0              1.0   

     Matrix      Method   Analyte     Unit  Result ResQualCode   MDL    RL  \
28   tissue  EPA 200.8m  Selenium  ug/g ww     NaN          ND  0.21  0.64   
29   tissue  EPA 200.8m  Selenium  ug/g ww     NaN          ND  0.21  0.64   
30   tissue  EPA 200.8m  Selenium  ug/g ww     NaN          ND  0.21  0.64   
31   tissue  EPA 200.8m  Selenium  ug/g ww     NaN          ND  0.21  0.64   
169  tissue    MPSL-120  Selenium  ug/g ww     NaN          ND  0.49  1.46   

    QACode BatchVerification ComplianceCode  DilutionFactor LabSampleID  \
28     NaN               VAC            Com             1.0  C122MISLOB   
29     NaN               VAC            Com             1.0  C122MISLOB   
30     NaN               VAC            Com             1.0  C122MISLOB   
31     NaN               VAC            Com             1.0  C122MISLOB   
169    NaN           VAC,VMD           Qual             1.0  23C1NRBTIL   

    ResultComments PrepPreservationName PrepPreservationDate  \
28             NaN          FieldFrozen           02/15/2023   
29             NaN          FieldFrozen           02/15/2023   
30             Na

In [108]:
results_null.shape

(91569, 117)

In [109]:
results_null['Analyte'].value_counts() # see how many values in each Analyte that has null results

Analyte
Selenium                                                 3100
PCB 169                                                   934
PCB 126                                                   821
Aldrin                                                    810
Heptachlor Epoxide                                        804
Chlordane, trans-                                         770
Chlordane, cis-                                           769
HCH, alpha-                                               766
Dieldrin                                                  765
Heptachlor                                                758
PCB AROCLOR 1260                                          758
PCB 027                                                   733
Endrin                                                    733
Mirex                                                     730
HCH, gamma-                                               726
HCH, beta-                                                722


In [110]:
df_tissue['Analyte'].value_counts() #looking at the total value counts of each analyte 

Analyte
Moisture                                                                 19663
Mercury                                                                  10732
Selenium                                                                  8719
Terphenyl-d14(Surrogate)                                                  2649
Lipid                                                                     2122
Age                                                                       1833
Fluorobiphenyl, 2-(Surrogate)                                             1222
DDE(p,p')                                                                 1074
PCB 052                                                                   1044
PCB 118                                                                   1042
PCB 170                                                                   1042
PCB 187                                                                   1040
PCB 066                                     

Looks like this dataset is only reliable for predicting mercury levels only since it doesn't have any null results. Determining PCBs and PBDEs levels may or may not work. I will need to group all subtypes of PCBs and PBDEs and see what the total ratio of null results there is. Since selenium has almost 40% of results with null values, this dataset is not a reliable source to determine the affinity between mercury level & selenium. Maybe I will need to add more years to my dataset... depending on how I want to analyze this data... let's keep going

In [111]:
df_tissue.head() # just a sanity check that I didn't break anything... yet

ProgramName  \
0  Surface Water Ambient Monitoring Program   
1  Surface Water Ambient Monitoring Program   
2  Surface Water Ambient Monitoring Program   
3  Surface Water Ambient Monitoring Program   
4  Surface Water Ambient Monitoring Program   

                          ParentProjectName            ProjectCode  \
0  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
1  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
3  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
4  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   

                                 ProjectName           CompositeID  \
0  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
1  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
2  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
3  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   
4  Statewide BMP Realignment RWB9 Study 2022  C1_90208OCNHRLN22LOB   

        StationName StationCode   Latitude   Longitude GeometryShape  \
0  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
1  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
2  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
3  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   
4  Oceanside Harbor   90208OCNH  33.209001 -117.401035   PolygonCent   

  EarliestDateSampled CompositeType                CommonName  \
0          04/19/2023        normal  California Spiny Lobster   
1          04/19/2023        normal  California Spiny Lobster   
2          04/19/2023        normal  California Spiny Lobster   
3          04/19/2023        normal  California Spiny Lobster   
4          04/19/2023        normal  California Spiny Lobster   

                 FinalID TissuePrep               TissueName  \
0  Panulirus interruptus        NaN  whole organism no shell   
1  Panulirus interruptus        NaN  whole organism no shell   
2  Panulirus interruptus        NaN  whole organism no shell   
3  Panulirus interruptus        NaN  whole organism no shell   
4  Panulirus interruptus        NaN  whole organism no shell   

   NumberFishperComp SampleTypeCode  CompositeReplicate  ResultReplicate  \
0                4.0      Composite                 1.0              1.0   
1                4.0      Composite                 1.0              1.0   
2                4.0      Composite                 1.0              1.0   
3                4.0      Composite                 1.0              1.0   
4                4.0      Composite                 1.0              1.0   

   Matrix      Method   Analyte     Unit  Result ResQualCode   MDL    RL  \
0  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
1  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
2  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
3  tissue  EPA 200.8m   Arsenic  ug/g ww    8.46           =  0.06  0.18   
4  tissue  EPA 200.8m  Moisture        %   75.50           =   NaN   NaN   

  QACode BatchVerification ComplianceCode  DilutionFactor LabSampleID  \
0    NaN               VAC            Com             1.0  C122OCHLOB   
1    NaN               VAC            Com             1.0  C122OCHLOB   
2    NaN               VAC            Com             1.0  C122OCHLOB   
3    NaN               VAC            Com             1.0  C122OCHLOB   
4    NaN               VAC            Com             1.0  C122OCHLOB   

  ResultComments PrepPreservationName PrepPreservationDate  \
0            NaN          FieldFrozen           04/19/2023   
1            NaN          FieldFrozen           04/19/2023   
2            NaN          FieldFrozen           04/19/2023   
3            NaN          FieldFrozen           04/19/2023   
4            NaN          FieldFroz

I want to explore the idea to see if there is a specific I want to focus on. Ideally I would love to focus on my local area, but looks like the majority of this data is concentrated on the San Diego region. (No sh*t sherlock :p, the world-famous Scripps Institude, and UCSD is right there, but still, kinda disappointed about the lack of data in the SF Bay Area...) Since there is only 220 stations, I decided to keep all. So the analysis will be on all of California.

In [112]:
df_tissue['StationName'].value_counts(normalize= True) # normalize = true returns the relative frequency

StationName
SD North Bay                                                                                            0.046583
Santa Cruz Pier                                                                                         0.042247
SD South Bay                                                                                            0.035628
Perris Reservoir                                                                                        0.031716
Peck Road Water Conservation Park                                                                       0.028787
San Diego Bay at Shelter Island East                                                                    0.026735
New River Outlet                                                                                        0.025383
Pillar Point Harbor                                                                                     0.023967
San Diego Bay at Spanish Landing                                                    

In [113]:
df_tissue['StationName'].nunique()

220

Next, I wanted to see if there is specific specie that I would like to focus on, since this is an exhaustive list of species. I picked Pacific Oyster  to look at first since it has the highest frequency.

In [114]:
pd.set_option('display.max_rows', None)
df_tissue['CommonName'].value_counts()

CommonName
Pacific Oyster                   42417
Largemouth Bass                  32262
Jacksmelt                        16436
Barred Surfperch                 14960
Common Carp                      10550
Tilapia spp.                     10087
Channel Catfish                   9577
Bluegill                          9464
Freshwater Clam                   6933
Spotted Sand Bass                 6778
Flathead Catfish                  6697
Shiner Surfperch                  5888
Northern Anchovy                  4960
California Mussel                 4599
Rainbow Trout                     3621
White Croaker                     2906
White Surfperch                   2241
Sacramento Sucker                 1961
Striped Mullet                    1930
Chub Mackerel                     1759
Spotted Bass                      1701
Leopard shark                     1561
Blue Catfish                      1544
California Corbina                1317
Threadfin Shad                    1285
Green Sunfish 

In [115]:
oyster= df_tissue[df_tissue['CommonName'] == 'Pacific Oyster'] # I filtered out the rows with Pacific Oyster and made another temp df
oyster.shape # to see how much of a sample size I am dealing with

(42417, 117)

In [116]:
oyster.head()# always sanity checking to see any new dfs I've made looks ok

ProgramName  \
2122  Surface Water Ambient Monitoring Program   
2123  Surface Water Ambient Monitoring Program   
2124  Surface Water Ambient Monitoring Program   
2125  Surface Water Ambient Monitoring Program   
2126  Surface Water Ambient Monitoring Program   

                             ParentProjectName            ProjectCode  \
2122  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2123  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2124  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2125  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   
2126  SWAMP Bioaccumulation Monitoring Program  SWB_Realign_RWB9_2022   

                                    ProjectName           CompositeID  \
2122  Statewide BMP Realignment RWB9 Study 2022  C1_90110DANARLN22OYS   
2123  Statewide BMP Realignment RWB9 Study 2022  C1_90110DANARLN22OYS   
2124  Statewide BMP Realignment RWB9 Study 2022  C1_90110DANARLN22OYS   
2125  Statewide BMP Realignment RWB9 Study 2022  C1_90110DANARLN22OYS   
2126  Statewide BMP Realignment RWB9 Study 2022  C1_90110DANARLN22OYS   

            StationName StationCode   Latitude   Longitude GeometryShape  \
2122  Dana Point Harbor   90110DANA  33.458265 -117.696586   PolygonCent   
2123  Dana Point Harbor   90110DANA  33.458265 -117.696586   PolygonCent   
2124  Dana Point Harbor   90110DANA  33.458265 -117.696586   PolygonCent   
2125  Dana Point Harbor   90110DANA  33.458265 -117.696586   PolygonCent   
2126  Dana Point Harbor   90110DANA  33.458265 -117.696586   PolygonCent   

     EarliestDateSampled CompositeType      CommonName            FinalID  \
2122          11/22/2022        normal  Pacific Oyster  Crassostrea gigas   
2123          11/22/2022        normal  Pacific Oyster  Crassostrea gigas   
2124          11/22/2022        normal  Pacific Oyster  Crassostrea gigas   
2125          11/22/2022        normal  Pacific Oyster  Crassostrea gigas   
2126          11/22/2022        normal  Pacific Oyster  Crassostrea gigas   

     TissuePrep                                   TissueName  \
2122        NaN  soft tissue (e.g. clams) with gonads intact   
2123        NaN  soft tissue (e.g. clams) with gonads intact   
2124        NaN  soft tissue (e.g. clams) with gonads intact   
2125        NaN  soft tissue (e.g. clams) with gonads intact   
2126        NaN  soft tissue (e.g. clams) with gonads intact   

      NumberFishperComp SampleTypeCode  CompositeReplicate  ResultReplicate  \
2122                1.0      Composite                 1.0              1.0   
2123                1.0      Composite                 1.0              1.0   
2124                1.0      Composite                 1.0              1.0   
2125                1.0      Composite                 1.0              1.0   
2126                1.0      Composite                 1.0              1.0   

      Matrix      Method   Analyte     Unit  Result ResQualCode    MDL    RL  \
2122  tissue  EPA 200.8m   Arsenic  ug/g dw   8.250           =  0.220  0.65   
2123  tissue  EPA 200.8m  Moisture        %  88.200           =    NaN   NaN   
2124  tissue  EPA 200.8m  Selenium  ug/g dw   4.320           =  0.780  2.35   
2125  tissue   EPA 7473M   Mercury  ug/g dw   0.043           =  0.009  0.03   
2126  tissue   EPA 7473M  Moisture        %  88.260           =    NaN   NaN   

     QACode BatchVerification ComplianceCode  DilutionFactor LabSampleID  \
2122    NaN               VAC            Com             1.0   2023-0027   
2123    NaN               VAC            Com             1.0   2023-0027   
2124    NaN               VAC            Com             1.0   2023-0027   
2125    NaN               VAC            Com             1.0   2023-0027   
2126    NaN               VAC            Com             1.0   2023-0027   

     ResultComments PrepPreservationName PrepPreservationDate  \
2122            NaN          FieldFrozen           11/22/2022   
2123    

In [117]:
oysterresults_null = oyster.loc[oyster['Result'].isnull()] #see how many null results there are... seems a lot
oysterresults_null.shape

(29867, 117)

In [118]:
oyster['Analyte'].value_counts() #see what kind of analyte is in oyster, no mercury

Analyte
Terphenyl-d14(Surrogate)                                          1120
Moisture                                                          1110
Fluorobiphenyl, 2-(Surrogate)                                      560
Mercury, Methyl                                                    492
Mercury                                                            307
Dicofol                                                            301
HCH, gamma-                                                        301
Endrin                                                             301
Endrin Aldehyde                                                    301
Endrin Ketone                                                      301
HCH, alpha-                                                        301
HCH, beta-                                                         301
HCH, delta-                                                        301
Heptachlor Epoxide                                                 30

Just looking at oyster doesn't seem like a good choice. Next, I want to look at the list of species that has at least 5000 samples of  

Pacific Oyster                   42417
Largemouth Bass                  32262
Jacksmelt                        16436
Barred Surfperch                 14960
Common Carp                      10550
Tilapia spp.                     10087
Channel Catfish                   9577
Bluegill                          9464
Freshwater Clam                   6933
Spotted Sand Bass                 6778
Flathead Catfish                  6697
Shiner Surfperch                  5888

In [119]:
#samples_list= ['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch']
#samples= df_tissue[df_tissue['CommonName'] == ['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch']] #this method didn't work. Trying ".isin" method
#samples.head() 

In [120]:
#samples_list= ['Pacific Oyster','Largemouth Bass', 'Jacksmelt']#'Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch']

#samples = df_tissue[df_tissue['CommonName'].isin(samples_list)]
#print(samples)# this didn't work, crashed the thing....

In [121]:
#samples_list= ['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch']


#for col in samples_list:
    #print(f"Shape of column '{col}': {df[col].shape}")

In [122]:
#samples= df_tissue[df_tissue.CommonName.str.isin(['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch'])]

In [123]:
#samples = np.where((df_tissue['CommonName']== ['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch']))
#display(df_tissue.loc[samples])

In [124]:
#species = ['CommonName'].isin(['Pacific Oyster','Largemouth Bass','Jacksmelt','Barred Surfperch','Common Carp','Tilapia spp.','Channel Catfish','Bluegill','Freshwater Clam','Spotted Sand Bass','Flathead Catfish','Shiner Surfperch'])

#samples= df_tissue[df_tissue.CommonName.str.isin('species')]

***Commented out code to save memory***

Leave that for now... next, I wanted to see if I can add water chemistry to the data.. this is my first source: https://data.ca.gov/dataset/surface-water-chemistry-results-ceden-augmentation/resource/e07c5e0b-cace-4b70-9f13-b3e696cd5a99 



In [125]:
H20=pd.read_csv('ceden-data-augmentation_chem-field_2024-05-06.csv')
H20.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/2340534810.py:1: DtypeWarning: Columns (10,13,14,19,24,25,26,27,28,29,30,39,41,42,43,44,45,47,48,49,50,67,72) have mixed types. Specify dtype option on import or set low_memory=False.
  H20=pd.read_csv('ceden-data-augmentation_chem-field_2024-05-06.csv')


Program             ParentProject  \
0  Proposition 84  Prop84-City of San Diego   
1  Proposition 84  Prop84-City of San Diego   
2  Proposition 84  Prop84-City of San Diego   
3  Proposition 84  Prop84-City of San Diego   
4  Proposition 84  Prop84-City of San Diego   

                                             Project StationName StationCode  \
0  City of San Diego Hypolimnetic Oxygenation System         HGA         HGA   
1  City of San Diego Hypolimnetic Oxygenation System         HGA         HGA   
2  City of San Diego Hypolimnetic Oxygenation System         HGA         HGA   
3  City of San Diego Hypolimnetic Oxygenation System         HGA         HGA   
4  City of San Diego Hypolimnetic Oxygenation System         HGA         HGA   

   SampleDate CollectionTime  LocationCode  CollectionDepth  \
0  2017-04-18          00:00  Not Recorded              0.0   
1  2017-04-18          00:00  Not Recorded              3.0   
2  2017-04-18          00:00  Not Recorded              6.0   
3  2017-04-18          00:00  Not Recorded              9.0   
4  2017-04-18          00:00  Not Recorded             12.0   

  UnitCollectionDepth SampleTypeCode  CollectionReplicate  ResultsReplicate  \
0                   m     Field_Grab                    1                 1   
1                   m     Field_Grab                    1                 1   
2                   m     Field_Grab                    1                 1   
3                   m     Field_Grab                    1                 1   
4                   m     Field_Grab                    1                 1   

                      LabBatch LabSampleID   MatrixName MethodName  \
0  UCMercedEAL_20170418_NO3NO2         NaN  samplewater  EPA 353.2   
1  UCMercedEAL_20170418_NO3NO2         NaN  samplewater  EPA 353.2   
2  UCMercedEAL_20170418_NO3NO2         NaN  samplewater  EPA 353.2   
3  UCMercedEAL_20170418_NO3NO2         NaN  samplewater  EPA 353.2   
4  UCMercedEAL_20170418_NO3NO2         NaN  samplewater  EPA 353.2   

                         Analyte  Unit Result  Observation   MDL    RL  \
0  Nitrate + Nitrite as N, Total  mg/L    NaN          NaN  0.05 -88.0   
1  Nitrate + Nitrite as N, Total  mg/L    NaN          NaN  0.05 -88.0   
2  Nitrate + Nitrite as N, Total  mg/L  0.492          NaN  0.05 -88.0   
3  Nitrate + Nitrite as N, Total  mg/L  0.662          NaN  0.05 -88.0   
4  Nitrate + Nitrite as N, Total  mg/L  0.759          NaN  0.05 -88.0   

  ResultQualCode QACode BatchVerification ComplianceCode SampleComments  \
0             ND    NaN                NR             NR            NaN   
1             ND    NaN                NR             NR            NaN   
2              =    NaN                NR             NR            NaN   
3              =    NaN                NR             NR            NaN   
4              =    NaN                NR             NR            NaN   

  CollectionComments ResultsComments BatchComments EventCode    ProtocolCode  \
0                NaN             NaN           NaN        WQ  Not Applicable   
1                NaN             NaN           NaN        WQ  Not Applicable   
2                NaN             NaN           NaN        WQ  Not Applicable   
3                NaN             NaN           NaN        WQ  Not Applicable   
4                NaN             NaN           NaN        WQ  Not Applicable   

    SampleAgency  GroupSamples CollectionMethodName  Latitude  Longitude  \
0  UC Merced EAL           NaN           Water_Grab  33.04581 -117.12715   
1  UC Merced EAL           NaN           Water_Grab  33.04581 -117.12715   
2  UC Merced EAL           NaN           Water_Grab  33.04581 -117.12715   
3  UC Merced EAL           NaN           Water_Grab  33.04581 -117.12715   
4  UC Merced EAL           NaN           Water_Grab  33.04581 -117.12715   

  CollectionDeviceDescription CalibrationDate PositionWaterColumn  \
0            Van Dorn Sampler             NaN      Not Applicable   

In [126]:
H20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1523413 entries, 0 to 1523412
Data columns (total 73 columns):
 #   Column                       Dtype  
---  ------                       -----  
 0   Program                      object 
 1   ParentProject                object 
 2   Project                      object 
 3   StationName                  object 
 4   StationCode                  object 
 5   SampleDate                   object 
 6   CollectionTime               object 
 7   LocationCode                 object 
 8   CollectionDepth              float64
 9   UnitCollectionDepth          object 
 10  SampleTypeCode               object 
 11  CollectionReplicate          int64  
 12  ResultsReplicate             int64  
 13  LabBatch                     object 
 14  LabSampleID                  object 
 15  MatrixName                   object 
 16  MethodName                   object 
 17  Analyte                      object 
 18  Unit                         object 
 19  

In [127]:
Stations_compare = df_tissue['StationName'].isin(H20['StationName']) #see how many values in "StationName" from the tissue dataset is in the H20 dataset

In [128]:
Stations_compare.value_counts() 

StationName
False    212870
True       3927
Name: count, dtype: int64

Too many false values, we can't joint these two

I looked into another datasource: https://data.ca.gov/dataset/surface-water-chemistry-results. The results are partitioned by years, so I had to go through the same process as I did to get df_tissue to get a master set. Then I will see if there is water chemistry data that are taken at the same stations at the same time as the entries in the df_tissue dataset

***Also commented out codes to save memory***

In [129]:
water23=pd.read_csv('waterchemistrydata_year-2023_2024-10-01.csv')
water23.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/2417328549.py:1: DtypeWarning: Columns (13,14,27,28,29,30,39,41,42,43,44,45,48,49,50,51,52,54,56,58,59,60,61,67,70) have mixed types. Specify dtype option on import or set low_memory=False.
  water23=pd.read_csv('waterchemistrydata_year-2023_2024-10-01.csv')


Program  \
0  Alvarado Creek Restoration   
1  Alvarado Creek Restoration   
2  Alvarado Creek Restoration   
3  Alvarado Creek Restoration   
4  Alvarado Creek Restoration   

                                     ParentProject  \
0  Alvarado Creek Bioassessment Monitoring Program   
1  Alvarado Creek Bioassessment Monitoring Program   
2  Alvarado Creek Bioassessment Monitoring Program   
3  Alvarado Creek Bioassessment Monitoring Program   
4  Alvarado Creek Bioassessment Monitoring Program   

                                          Project  \
0  Alvarado Creek Bioassessment Monitoring - 2023   
1  Alvarado Creek Bioassessment Monitoring - 2023   
2  Alvarado Creek Bioassessment Monitoring - 2023   
3  Alvarado Creek Bioassessment Monitoring - 2023   
4  Alvarado Creek Bioassessment Monitoring - 2023   

                              StationName StationCode           SampleDate  \
0  Alvarado Creek, lower SDSU restoration   907ALVLWR  2023-06-12 00:00:00   
1  Alvarado Creek, lower SDSU restoration   907ALVLWR  2023-06-12 00:00:00   
2  Alvarado Creek, lower SDSU restoration   907ALVLWR  2023-06-12 00:00:00   
3  Alvarado Creek, upper SDSU restoration   907ALVUPR  2023-06-13 00:00:00   
4  Alvarado Creek, upper SDSU restoration   907ALVUPR  2023-06-13 00:00:00   

        CollectionTime LocationCode  CollectionDepth UnitCollectionDepth  \
0  1899-12-30 09:45:00            X            -88.0                   m   
1  1899-12-30 09:45:00            X            -88.0                   m   
2  1899-12-30 09:45:00            X              0.1                   m   
3  1899-12-30 09:40:00            X            -88.0                   m   
4  1899-12-30 09:40:00            X              0.1                   m   

  SampleTypeCode  CollectionReplicate  ResultsReplicate LabBatch LabSampleID  \
0   Not Recorded                  7.0               1.0      NaN         NaN   
1   Not Recorded                 20.0               1.0      NaN         NaN   
2   Not Recorded                  1.0               1.0      NaN         NaN   
3   Not Recorded                 19.0               1.0      NaN         NaN   
4   Not Recorded                  1.0               1.0      NaN         NaN   

    MatrixName     MethodName                   Analyte  Unit  Result  \
0  samplewater  Velocity Area                  Velocity  ft/s    1.25   
1  samplewater  Velocity Area                  Velocity  ft/s    0.28   
2  samplewater   FieldMeasure                        pH  none    7.84   
3  samplewater  Velocity Area                  Velocity  ft/s    0.37   
4  samplewater   FieldMeasure  Oxygen, Dissolved, Total  mg/L    7.02   

   Observation  MDL  RL ResultQualCode QACode BatchVerification  \
0          NaN  NaN NaN              =    NaN                NR   
1          NaN  NaN NaN              =    NaN                NR   
2          NaN  NaN NaN              =    NaN                NR   
3          NaN  NaN NaN              =    NaN                NR   
4          NaN  NaN NaN              =    NaN                NR   

  ComplianceCode                                     SampleComments  \
0           Pend  Flow has considerably decreased since the morn...   
1           Pend  Flow has considerably decreased since the morn...   
2           Pend  Flow has considerably decreased since the morn...   
3           Pend  slope previously determined and has not signif...   
4           Pend  slope previously determined and has not signif...   

  CollectionComments ResultsComments BatchComments EventCode   ProtocolCode  \
0                NaN             NaN           NaN        BA  SWAMP_2016_WS   
1                NaN             NaN           NaN        BA  SWAMP_2016_WS   
2                NaN             NaN           NaN        BA  SWAMP_2016_WS   
3                NaN             NaN           NaN        BA  SWAMP_2016_WS   
4                NaN             NaN           NaN        BA  SWAMP_2016_WS   

     SampleAgency  GroupSamp

In [130]:
water22=pd.read_csv('waterchemistrydata_year-2022_2024-10-01.csv')
water22.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/678843278.py:1: DtypeWarning: Columns (13,14,24,27,28,29,30,39,41,42,43,44,45,48,49,50,51,52,54,56,58,59,60,61,67) have mixed types. Specify dtype option on import or set low_memory=False.
  water22=pd.read_csv('waterchemistrydata_year-2022_2024-10-01.csv')


Program  \
0  Alvarado Creek Restoration   
1  Alvarado Creek Restoration   
2  Alvarado Creek Restoration   
3  Alvarado Creek Restoration   
4  Alvarado Creek Restoration   

                                     ParentProject  \
0  Alvarado Creek Bioassessment Monitoring Program   
1  Alvarado Creek Bioassessment Monitoring Program   
2  Alvarado Creek Bioassessment Monitoring Program   
3  Alvarado Creek Bioassessment Monitoring Program   
4  Alvarado Creek Bioassessment Monitoring Program   

                                          Project  \
0  Alvarado Creek Bioassessment Monitoring - 2022   
1  Alvarado Creek Bioassessment Monitoring - 2022   
2  Alvarado Creek Bioassessment Monitoring - 2022   
3  Alvarado Creek Bioassessment Monitoring - 2022   
4  Alvarado Creek Bioassessment Monitoring - 2022   

                                     StationName StationCode  \
0         Alvarado Creek, lower SDSU restoration   907ALVLWR   
1         Alvarado Creek, lower SDSU restoration   907ALVLWR   
2         Alvarado Creek, upper SDSU restoration   907ALVUPR   
3         Alvarado Creek, upper SDSU restoration   907ALVUPR   
4  Alvarado Creek, upstream of pedestrian bridge   907ALVPED   

            SampleDate       CollectionTime LocationCode  CollectionDepth  \
0  2022-06-21 00:00:00  1899-12-30 11:00:00            X            -88.0   
1  2022-06-21 00:00:00  1899-12-30 11:00:00            X              0.1   
2  2022-06-22 00:00:00  1899-12-30 10:00:00            X            -88.0   
3  2022-06-22 00:00:00  1899-12-30 10:00:00            X              0.1   
4  2022-06-23 00:00:00  1899-12-30 10:20:00            X            -88.0   

  UnitCollectionDepth SampleTypeCode  CollectionReplicate  ResultsReplicate  \
0                   m   Not Recorded                 19.0               1.0   
1                   m   Not Recorded                  1.0               1.0   
2                   m   Not Recorded                 20.0               1.0   
3                   m   Not Recorded                  1.0               1.0   
4                   m   Not Recorded                 17.0               1.0   

  LabBatch LabSampleID   MatrixName     MethodName  \
0      NaN         NaN  samplewater  Velocity Area   
1      NaN         NaN  samplewater   FieldMeasure   
2      NaN         NaN  samplewater  Velocity Area   
3      NaN         NaN  samplewater   FieldMeasure   
4      NaN         NaN  samplewater  Velocity Area   

                       Analyte   Unit   Result  Observation  MDL  RL  \
0                     Velocity   ft/s     0.41          NaN  NaN NaN   
1    Oxygen, Saturation, Total      %    81.50          NaN  NaN NaN   
2                     Velocity   ft/s     0.04          NaN  NaN NaN   
3  SpecificConductivity, Total  uS/cm  3347.00          NaN  NaN NaN   
4                     Velocity   ft/s     0.05          NaN  NaN NaN   

  ResultQualCode QACode BatchVerification ComplianceCode  \
0              =    NaN                NR           Pend   
1              =    NaN                NR           Pend   
2              =    NaN                NR           Pend   
3              =    NaN                NR           Pend   
4              =    NaN                NR           Pend   

                                      SampleComments CollectionComments  \
0  Saw sunfish, orange/white fish, mosquito fish,...                NaN   
1  Saw sunfish, orange/white fish, mosquito fish,...                NaN   
2                                                NaN                NaN   
3                                                NaN                NaN   
4                                                NaN                NaN   

  ResultsComments BatchComments EventCode   ProtocolCode    SampleAgency  \
0             NaN           NaN        BA  SWAMP_2016_WS  ICF-Sacramento   
1             NaN           NaN        BA  SWAMP_2016_WS  ICF-Sacramento   
2             NaN           NaN        BA  SWAMP_2016_WS  

In [131]:
water21=pd.read_csv('waterchemistrydata_year-2021_2024-10-01.csv')
water21.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/3630598429.py:1: DtypeWarning: Columns (13,14,24,27,28,29,30,39,41,42,43,44,45,48,49,50,51,52,54,56,58,59,60,61,67) have mixed types. Specify dtype option on import or set low_memory=False.
  water21=pd.read_csv('waterchemistrydata_year-2021_2024-10-01.csv')


Program  \
0  Alvarado Creek Human Source Investigation   
1  Alvarado Creek Human Source Investigation   
2  Alvarado Creek Human Source Investigation   
3  Alvarado Creek Human Source Investigation   
4  Alvarado Creek Human Source Investigation   

                               ParentProject  \
0  Alvarado Creek Human Source Investigation   
1  Alvarado Creek Human Source Investigation   
2  Alvarado Creek Human Source Investigation   
3  Alvarado Creek Human Source Investigation   
4  Alvarado Creek Human Source Investigation   

                                     Project  \
0  Alvarado Creek Human Source Investigation   
1  Alvarado Creek Human Source Investigation   
2  Alvarado Creek Human Source Investigation   
3  Alvarado Creek Human Source Investigation   
4  Alvarado Creek Human Source Investigation   

                        StationName  StationCode           SampleDate  \
0           City MS4 Outfall DW1063       DW1063  2021-02-09 00:00:00   
1           City MS4 Outfall DW1063       DW1063  2021-02-09 00:00:00   
2           City MS4 Outfall DW1063       DW1063  2021-02-10 00:00:00   
3           City MS4 Outfall DW1063       DW1063  2021-02-11 00:00:00   
4  Proxy to City MS4 Outfall DW1132  DW1132-UP01  2021-02-10 00:00:00   

        CollectionTime  LocationCode  CollectionDepth UnitCollectionDepth  \
0  1899-12-30 10:20:00  Not Recorded             0.01                   m   
1  1899-12-30 10:20:00  Not Recorded             0.01                   m   
2  1899-12-30 10:25:00  Not Recorded             0.01                   m   
3  1899-12-30 09:45:00  Not Recorded             0.01                   m   
4  1899-12-30 08:22:00  Not Recorded             0.01                   m   

  SampleTypeCode  CollectionReplicate  ResultsReplicate LabBatch LabSampleID  \
0   Not Recorded                  1.0               1.0      NaN         NaN   
1   Not Recorded                  1.0               1.0      NaN         NaN   
2   Not Recorded                  1.0               1.0      NaN         NaN   
3   Not Recorded                  1.0               1.0      NaN         NaN   
4   Not Recorded                  1.0               1.0      NaN         NaN   

  MatrixName    MethodName                          Analyte   Unit  Result  \
0     runoff  FieldMeasure                               pH   none   7.650   
1     runoff  FieldMeasure                      Temperature  Deg C  17.400   
2     runoff  FieldMeasure                               pH   none   7.860   
3     runoff  FieldMeasure  Chlorine, Total Residual, Total   mg/L   0.135   
4     runoff  FieldMeasure                               pH   none   7.700   

   Observation  MDL  RL ResultQualCode QACode BatchVerification  \
0          NaN  NaN NaN              =    NaN                NR   
1          NaN  NaN NaN              =    NaN                NR   
2          NaN  NaN NaN              =    NaN                NR   
3          NaN  NaN NaN              =    NaN                NR   
4          NaN  NaN NaN              =    NaN                NR   

  ComplianceCode SampleComments CollectionComments ResultsComments  \
0             NR            NaN                NaN             NaN   
1             NR            NaN                NaN             NaN   
2             NR            NaN                NaN             NaN   
3             NR            NaN                NaN             NaN   
4             NR            NaN                NaN             NaN   

  BatchComments EventCode  ProtocolCode SampleAgency  GroupSamples  \
0           NaN        WQ  Not Recorded    WoodGroup           NaN   
1           NaN        WQ  Not Recorded    WoodGroup           NaN   
2           NaN        WQ  Not Recorded    WoodGroup           NaN   
3           NaN        WQ  Not Recorded    WoodGroup           NaN   
4           NaN        WQ  Not Recorded    WoodGroup           NaN   

  CollectionMethodName   Latitude   Longitude CollectionDeviceDescription

In [132]:
water20=pd.read_csv('waterchemistrydata_year-2020_2024-10-01.csv')
water20.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/3918048944.py:1: DtypeWarning: Columns (13,14,27,28,29,30,39,41,42,43,44,45,48,49,50,51,52,54,56,58,59,60,61,67) have mixed types. Specify dtype option on import or set low_memory=False.
  water20=pd.read_csv('waterchemistrydata_year-2020_2024-10-01.csv')


Program                                 ParentProject  \
0  401 Certification  San Diego Region Water Quality Control Board   
1  401 Certification  San Diego Region Water Quality Control Board   
2  401 Certification  San Diego Region Water Quality Control Board   
3  401 Certification  San Diego Region Water Quality Control Board   
4  401 Certification  San Diego Region Water Quality Control Board   

                                       Project  \
0  Discovery Village Bioassessment Precon 2020   
1  Discovery Village Bioassessment Precon 2020   
2  Discovery Village Bioassessment Precon 2020   
3  Discovery Village Bioassessment Precon 2020   
4  Discovery Village Bioassessment Precon 2020   

                                     StationName StationCode  \
0  Discovery Village Downstream San Marcos Creek       DV002   
1  Discovery Village Downstream San Marcos Creek       DV002   
2  Discovery Village Downstream San Marcos Creek       DV002   
3  Discovery Village Downstream San Marcos Creek       DV002   
4    Discovery Village Upstream San Marcos Creek       DV001   

            SampleDate       CollectionTime LocationCode  CollectionDepth  \
0  2020-07-15 00:00:00  1899-12-30 08:13:00            X            -88.0   
1  2020-07-15 00:00:00  1899-12-30 08:13:00            X            -88.0   
2  2020-07-15 00:00:00  1899-12-30 08:13:00            X            -88.0   
3  2020-07-15 00:00:00  1899-12-30 08:13:00            X            -88.0   
4  2020-07-15 00:00:00  1899-12-30 14:20:00            X            -88.0   

  UnitCollectionDepth SampleTypeCode  CollectionReplicate  ResultsReplicate  \
0                   m   Not Recorded                  3.0               1.0   
1                   m   Not Recorded                  4.0               1.0   
2                   m   Not Recorded                  8.0               1.0   
3                   m   Not Recorded                  9.0               1.0   
4                   m   Not Recorded                  2.0               1.0   

  LabBatch LabSampleID   MatrixName     MethodName   Analyte  Unit  Result  \
0      NaN         NaN  samplewater  Velocity Area  Velocity  ft/s     0.2   
1      NaN         NaN  samplewater  Velocity Area  Velocity  ft/s     0.4   
2      NaN         NaN  samplewater  Velocity Area  Velocity  ft/s     0.1   
3      NaN         NaN  samplewater  Velocity Area  Velocity  ft/s     0.2   
4      NaN         NaN  samplewater  Velocity Area  Velocity  ft/s     0.1   

   Observation  MDL  RL ResultQualCode QACode BatchVerification  \
0          NaN  NaN NaN              =    NaN                NR   
1          NaN  NaN NaN              =    NaN                NR   
2          NaN  NaN NaN              =    NaN                NR   
3          NaN  NaN NaN              =    NaN                NR   
4          NaN  NaN NaN              =    NaN                NR   

  ComplianceCode                    SampleComments CollectionComments  \
0           Pend  No cobbles detected along reach.                NaN   
1           Pend  No cobbles detected along reach.                NaN   
2           Pend  No cobbles detected along reach.                NaN   
3           Pend  No cobbles detected along reach.                NaN   
4           Pend  No cobbles present in the reach.                NaN   

  ResultsComments BatchComments EventCode   ProtocolCode SampleAgency  \
0             NaN           NaN        BA  SWAMP_2016_WS    MERKELINC   
1             NaN           NaN        BA  SWAMP_2016_WS    MERKELINC   
2             NaN           NaN        BA  SWAMP_2016_WS    MERKELINC   
3             NaN           NaN        BA  SWAMP_2016_WS    MERKELINC   
4             NaN           NaN        BA  SWAMP_2016_WS    MERKELINC   

   GroupSamples CollectionMethodName  Latitude  Longitude  \
0           NaN         Field Method  33.13207 -117.18320   
1           NaN         Field Method  33.13207 -117.18320   
2           NaN         Field Metho

In [133]:
water19=pd.read_csv('waterchemistrydata_year-2019_2024-10-01.csv')
water19.head()

/var/folders/rr/4wp1syc10818wgtwwcq7_nvw0000gn/T/ipykernel_2647/1728294121.py:1: DtypeWarning: Columns (13,14,27,28,29,30,38,39,41,42,43,44,45,48,49,50,51,52,54,56,58,59,60,61,67,70) have mixed types. Specify dtype option on import or set low_memory=False.
  water19=pd.read_csv('waterchemistrydata_year-2019_2024-10-01.csv')


Program          ParentProject  \
0  American Rivers Restoration  Marsh Creek Watershed   
1  American Rivers Restoration  Marsh Creek Watershed   
2  American Rivers Restoration  Marsh Creek Watershed   
3  American Rivers Restoration  Marsh Creek Watershed   
4  American Rivers Restoration  Marsh Creek Watershed   

                    Project     StationName StationCode           SampleDate  \
0  Three Creeks Restoration  Creekside Park        MC13  2019-01-01 00:00:00   
1  Three Creeks Restoration  Creekside Park        MC13  2019-02-12 00:00:00   
2  Three Creeks Restoration  Creekside Park        MC13  2019-02-12 00:00:00   
3  Three Creeks Restoration  Creekside Park        MC13  2019-02-12 00:00:00   
4  Three Creeks Restoration  Creekside Park        MC13  2019-03-12 00:00:00   

        CollectionTime LocationCode  CollectionDepth UnitCollectionDepth  \
0  1899-12-30 09:15:00         Bank              0.1                   m   
1  1899-12-30 09:05:00         Bank              0.1                   m   
2  1899-12-30 09:05:00         Bank              0.1                   m   
3  1899-12-30 09:05:00         Bank              0.1                   m   
4  1899-12-30 08:59:00         Bank              0.1                   m   

  SampleTypeCode  CollectionReplicate  ResultsReplicate LabBatch LabSampleID  \
0   Not Recorded                  1.0               1.0      NaN         NaN   
1   Not Recorded                  1.0               1.0      NaN         NaN   
2   Not Recorded                  1.0               1.0      NaN         NaN   
3   Not Recorded                  1.0               1.0      NaN         NaN   
4   Not Recorded                  1.0               1.0      NaN         NaN   

    MatrixName    MethodName                      Analyte   Unit  Result  \
0  samplewater  FieldMeasure             Turbidity, Total    NTU   58.90   
1  samplewater  FieldMeasure     Oxygen, Dissolved, Total   mg/L   12.39   
2  samplewater  FieldMeasure                  Temperature  Deg C    6.53   
3  samplewater  FieldMeasure             Turbidity, Total    NTU   22.90   
4  samplewater  FieldMeasure  SpecificConductivity, Total  uS/cm  645.00   

   Observation  MDL  RL ResultQualCode QACode BatchVerification  \
0          NaN  NaN NaN              =    NaN                NR   
1          NaN  NaN NaN              =    NaN                NR   
2          NaN  NaN NaN              =    NaN                NR   
3          NaN  NaN NaN              =    NaN                NR   
4          NaN  NaN NaN              =    NaN                NR   

  ComplianceCode SampleComments CollectionComments ResultsComments  \
0             NR            NaN                NaN             NaN   
1             NR            NaN                NaN             NaN   
2             NR            NaN                NaN             NaN   
3             NR            NaN                NaN             NaN   
4             NR            NaN                NaN             NaN   

  BatchComments EventCode  ProtocolCode SampleAgency  GroupSamples  \
0           NaN        WQ  3CREEKS AMMP           AR           NaN   
1           NaN        WQ  3CREEKS AMMP           AR           NaN   
2           NaN        WQ  3CREEKS AMMP           AR           NaN   
3           NaN        WQ  3CREEKS AMMP           AR           NaN   
4           NaN        WQ  3CREEKS AMMP           AR           NaN   

  CollectionMethodName  Latitude  Longitude CollectionDeviceDescription  \
0         Field Method    37.923   -121.712                   Hach 2100   
1         Field Method    37.923   -121.712                 YSI 556 MPS   
2         Field Method    37.923   -121.712                 YSI 556 MPS   
3         Field Method    37.923   -121.712                   Hach 2100   
4         Field Method    37.923   -121.712                 YSI 556 MPS   

       CalibrationDate PositionWaterColumn PrepPreservationName  \
0  1950-01-01 00:00:00          subsurfac

In [134]:
using the multidf_shape function I have created

multidf_shape([water19,water20,water21,water22,water23]) 

SyntaxError: invalid syntax (493810215.py, line 1)

In [ ]:
using the compare_headers function I have created

compare_headers([water19,water20,water21,water22,water23]) 

In [ ]:
water19.columns.tolist() # to see a view of all columns that I don't need and drop them

In [ ]:
dataframes_water = [water19,water20,water21,water22,water23] #dropping these columns since I don't need the information to make the data sets smaller and make it easier to concat
for df in dataframes_water:
    df.drop(['Program','ParentProject','Project','CollectionTime','LocationCode','CollectionDepth','UnitCollectionDepth','SampleTypeCode','CollectionReplicate','ResultsReplicate','LabBatch','LabSampleID','MatrixName','MethodName','Observation','MDL','RL','ResultQualCode','QACode','BatchVerification','ComplianceCode','SampleComments','CollectionComments','ResultsComments','BatchComments','EventCode','ProtocolCode','SampleAgency','GroupSamples','CollectionMethodName','CollectionDeviceDescription','CalibrationDate','PositionWaterColumn','PrepPreservationName','PrepPreservationDate','DigestExtractMethod','DigestExtractDate','AnalysisDate','DilutionFactor','ExpectedValue','LabAgency','SubmittingAgency','SubmissionCode','OccupationMethod','StartingBank','DistanceFromBank','UnitDistanceFromBank','StreamWidth','UnitStreamWidth','StationWaterDepth','UnitStationWaterDepth','HydroMod','HydroModLoc','LocationDetailWQComments','ChannelWidth','UpstreamLength','DownStreamLength','TotalReach','LocationDetailBAComments','SampleID','DataQuality','DataQualityIndicator','Datum'], axis=1, inplace=True)

In [ ]:
df_water = pd.concat([water19,water20,water21,water22,water23])

In [ ]:
df_water.head()

In [ ]:
df_tissue.head() # sanity check

In [ ]:
df_tissue['StationName'].isin(df_water['StationName']).value_counts() # seeing if values in df_tissue under Sation Name is in df_water

In [ ]:
df_tissue['StationName'].unique()

In [ ]:
#Tissue_Stations = set(df_tissue['StationName'].unique())
#Water_Stations= set(df_water['StationName'].unique())

#missing_in_Water_Stations = Tissue_Stations - Water_Stations
#missing_in_Water_Stations

Again, too many false values again, meaning it's too many missing values. At this point, I think it's best to just look more into the df_tissue data set to see what I can correlate and predict

In [ ]:
df_tissue.groupby(['CommonName','Analyte'])['Result'].sum().sort_values(ascending=False)

In [ ]:
df_tissue.groupby(['StationName'])['CommonName'].value_counts()

I am going to save the masterset as a csv and reload it into the notebook for cleaning. 

In [ ]:
df_tissue.to_csv('Tissue_dataset_2019_2023.csv', index=False) # makes it not save an index col

In [ ]:
df = pd.read_csv('Tissue_dataset_2019_2023.csv')

In [ ]:
multidf_shape([df_tissue, df]) # make sure they are the same shapes

In [ ]:
compare_headers([df_tissue, df])

It looks like the data was saved as a CSV correctly, we can use it for our next phrase of data cleaning and EDA in the next notebook.